### PR TITLE
[UR][L0] Make urPlatformGetBackendOption return -ze-opt-level=2 for -O1 and -O2

### PR DESCRIPTION
**Pull request by sarnex**
IGC only accepts three values for -ze-opt-level, 0, 1 and 2, however dpcpp has at least four options, O0, O1, O2 and O3.

Right now we decided to map O1 and O2 to -ze-opt-level=1, but we have determined that this is not the best for user experience and may defy user expectation, and actually produces worse code than specifying nothing because IGC uses ze-opt-level=2 if nothing is specified.

https://github.com/intel/llvm/pull/12023

### CODE REVIEW

**Comment by sarnex** [comment]
@kbenzie Do you mind approving the CI run for this? Thanks!

**Review by rakeshkrishnaiyer — APPROVED** [review]
lgtm

**Comment by jandres742** [comment]
thanks @sarnex . could you open a PR in intel/llvm to validate the changes before merging? please see guidelines here https://oneapi-src.github.io/unified-runtime/core/CONTRIB.html#adapter-change-process and a sample here 

github.com/oneapi-src/unified-runtime/pull/1099
github.com/intel/llvm/pull/11958

**Comment by jandres742** [comment]
changes are looking good. We will wait for llvm testing.

**Comment by sarnex** [comment]
@jandres742 Sorry about that, the contribution guidelines changed since I last worked here and I assumed I didn't need to check :)

The llvm PR CI passed, so we should be good there

**Review by jandres742 — APPROVED** [review]

**Comment by jandres742** [comment]
thanks @sarnex 

@kbenzie : please merge when possible. Not mandatory 0.8.x.

**Comment by fabiomestre** [comment]
I have updated the target branch of this PR from the `adapters` branch to the `main` branch.
Development in UR is moving back to `main`. The `adapters` branch will soon be deleted.

**Comment by sarnex** [comment]
Sure, hopefully we can merge this soon, let me know if something is needed from me

**Comment by kbenzie** [comment]
> @fabiomestre @kbenzie Any ETA on this being merged? Thanks

We're aiming to clear the [ready to merge](https://github.com/oneapi-src/unified-runtime/pulls?q=is%3Aopen+is%3Apr+label%3A%22ready+to+merge%22) backlog this week.

**Comment by sarnex** [comment]
@kbenzie Cool thanks for the update

**Pull request merged**
